benches/benches/generate_test_data.rs

```diff
@@ -0,0 +1,120 @@
+use criterion::{
+    black_box,
+    criterion_group,
+    criterion_main,
+    Criterion,
+    Throughput,
+};
+use digest::Digest;
+use rand::{
+    rngs::StdRng,
+    Rng,
+    SeedableRng,
+};
+use sha2::Sha256;
+use std::iter::successors;
+
+fn random_bytes32<R>(rng: &mut R) -> [u8; 32]
+where
+    R: Rng + ?Sized,
+{
+    let mut bytes = [0u8; 32];
+    rng.fill(bytes.as_mut());
+    bytes
+}
+
+fn hash(bytes: &[u8]) -> [u8; 32] {
+    let mut hash = Sha256::new();
+    hash.update(bytes);
+    hash.finalize().into()
+}
+
+const UNIT_COEF: f64 = 100.0;
+
+fn busy<R>(coefficient: f64, rng: &mut R)
```

Comment:
I'm curious why you went with something that warms the CPU here. From what I'm reading I assume this is meant to be some arbitrary process. I would usually choose to do a sleep--or better something that mocks time.